halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/KubectlDeployer.java

```diff
@@ -122,4 +122,36 @@ public void flushInfrastructureCaches(KubectlServiceProvider serviceProvider,
       SpinnakerRuntimeSettings runtimeSettings) {
     throw new UnsupportedOperationException("todo(lwander)");
   }
+
+  @Override
+  public void deleteDisabledServices(KubectlServiceProvider serviceProvider,
+      AccountDeploymentDetails<KubernetesAccount> deploymentDetails,
+      GenerateService.ResolvedConfiguration resolvedConfiguration,
+      List<SpinnakerService.Type> serviceTypes) {
+
+    List<KubernetesV2Service> services = serviceProvider.getServicesByPriority(serviceTypes);
+    services.stream().forEach((service) -> {
+      if (service instanceof SidecarService) {
+        return;
+      }
+
+      ServiceSettings settings = resolvedConfiguration.getServiceSettings((SpinnakerService) service);
+      if (settings == null) {
+        return;
+      }
+
+      if (settings.getEnabled() == null || settings.getEnabled()) {
```

Comment:
Can you verify that this doesn't delete a service that was omitted by passing in `--service-names` for a deployment?

Reply:
verified

Reply:
Services provided in the service-names flag will be used in a list of services to be examined. The enabled flag is set/not set by the ServiceProvider later on.